Compiler/CLI/Options.cpp

```diff
@@ -109,8 +109,9 @@ void Options::configureOutPath() {
     if (pack() && outPath_.empty()) {
         if (standalone()) {
             outPath_ = mainFile_;
-            if (mainFile_.size() > 7) {
-                outPath_.resize(mainFile_.size() - 7);
+            std::size_t indexOfExtension = mainFile_.find_last_of(".");
+            if (indexOfExtension > 0) {
```

Comment:
Can you add a test against `std::string::npos` here please? (In case the path does not contain a full stop.)

Reply:
No problem for `std::string(llvm::sys::path::stem(mainFile_))` just, currently the code accepts "extension only" file, so for instance a file named `.emojic` can be compiled with `emojicodec .emojic` and the binary produced will overwrite `.emojic`. Is this the intended way ? If so I can do it, but I can also update the code to impose an input filename of *at least* one character.

Reply:
Yes, please add a restriction to at least one character. A file named just `.emojic` is not useful. We should check that in Package.cpp, though, shouldn't we?